policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/request/PolicyResponse.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.policy.request;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.resource.Resource;
+import uk.gov.gchq.palisade.rule.Rules;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+
+/**
+ * Represents the original data that has been sent from the client to Palisade Service for a request to access data.
```

Comment:
I thing starting lots of the message type comments with this line is confusing, I thing the comments should point out where this class sits in the chain. E.g this is the output from the policy service which would then be read by the query scope service. but I don't think we need to explain the end to end process. That is for other documentation to do. We should also be mentioning the paired classes (response and request). I don't think the Data Request object is relevant here as the query scope service should have manipulated the data before it is sent so it won't actually contain the same data